lib/diagnostic.rb

```diff
@@ -2,41 +2,42 @@
 
 # Ruby enumerable diagnostic
 class Diagnostic
+  include Enumerable
   # Question 1
   # Write ruby code to return the longest entry in the list supplied.
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.max_by(&:length)
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.any? {|num| num%2 != 0}
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    numbers.all? {|num| num%2 == 0}
   end
 
   # Question 5
   # Write ruby code that returns an array with all of the
   # entries from the list of words capitalized (i.e. 'foo' becomes 'Foo').
   def transform(words)
-    # Your code goes here
+    p words.entries {|word| "#{word[0].upcase}#{word[1, word.length]}"}
   end
```

Comment:
a few things to think about:
1) what was your expectation for the behavior of `.entries`? take a look at the documentation again: http://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-entries
2) in the block of code that comes after, what is `word`? what did you want it to be, and what is it in the code you wrote? 
3) assuming that you are getting the characters of a word by putting the index/range in the `[]`s, what is the value of `word.length`? If you are starting at index 1, what would be the appropriate length from there to the end of the string?